server/game/cards/02_SHD/units/SurvivorsGauntlet.ts

```diff
@@ -0,0 +1,45 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { RelativePlayer, TargetMode, WildcardCardType, WildcardRelativePlayer, WildcardZoneName } from '../../../core/Constants';
+
+export default class SurvivorsGauntlet extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '1503633301',
+            internalName: 'survivors-gauntlet',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addTriggeredAbility({
+            title: 'Attach an upgrade on a unit to another eligible unit controlled by the same player',
+            when: {
+                onCardPlayed: (event, context) => event.card === context.source,
+                onAttackDeclared: (event, context) => event.attack.attacker === context.source,
+            },
+            targetResolvers: {
+                chooseUpgrade: {
+                    mode: TargetMode.Single,
+                    cardTypeFilter: WildcardCardType.Upgrade,
+                    zoneFilter: WildcardZoneName.AnyArena,
+                    controller: WildcardRelativePlayer.Any,
+                    optional: true,
+                },
+                chooseUnit: {
+                    dependsOn: 'chooseUpgrade',
+                    mode: TargetMode.Single,
+                    cardTypeFilter: WildcardCardType.Unit,
+                    zoneFilter: WildcardZoneName.AnyArena,
+                    controller: (context) => (context.source.controller === context.targets.chooseUpgrade.controller ? RelativePlayer.Self : RelativePlayer.Opponent),
+                    cardCondition: (card, context) => context.targets.chooseUpgrade.isUpgrade() && context.targets.chooseUpgrade.parentCard !== card,
```

Comment:
Also needs to check upgrade eligibility, since some upgrades have restrictions like not being attachable to vehicles. I think it should be enough to just call `context.targets.chooseUpgrade.canAttach()`

Reply:
That's not really necessary because `AttachUpgradeSystem` will already call it in its implementation. This is already covered in the test, for example when moving `Frozen in Carbonite` the deployed leader is not selectable.